tools/projmgr/docs/Manual/Overview.md

```diff
@@ -0,0 +1,1317 @@
+
+The picture below shows a system that is composed of:
+
+- Project A: that implements a time-critical control algorithm running on a independent processor #2.
+- Project B: which is a diagram of a cloud connected IoT application with Machine Learning (ML) functionality.
+- Project C: that is the data model of the Machine Learning algorithm and separate to allow independent updates.
+- Project D: that implements the device security (for example with TF-M that runs with TrustZone in secure mode).
+
+In addition such systems may have a boot-loader that can be also viewed as another independent project.
+
+![Related Projects of an Embedded System](./images/Solution.png "Related Projects of an Embedded System")
+
+To manage the complexity of such related a projects, the `*.csolution.yml` file is introduced. At this level the `target-types` and `build-types` may be managed, so that a common set is available across the system. However it should be also possible to add project specific `build-types` at project level.  (tdb: `target-types` might be only possible at solution level).
+
+- `target-types` describe a different hardware target system and have therefore different API files for peripherals or a different hardware configuration.
+
+- `build-types` describe a build variant of the same hardware target. All `build-types` share the same API files for peripherals and the same hardware configuration, but may compile a different variant (i.e. with test I/O enabled) of an application.
+
+**Related Projects `iot-product.csolution.yml`**
+
+```yml
+target-types:
+  - type: Board
+    board: NUCLEO-L552ZE-Q
+
+  - type: Production-HW
+    device: STM32U5X          # specifies device
+      
+build-types:
+  - type: Debug
+    optimize: debug
+    debug: on
+
+  - type: Test
+    optimize: max
+    debug: on
+    
+solution:
+  - project: /security/TFM.cproject.yml
+    type: .Release
+  - project: /application/MQTT_AWS.cproject.yml
+  - project: /bootloader/Bootloader.cproject.yml
+    not-for-type: +Virtual
+```
+
+# Project Structure
+
+## Directory Structure  (old)
+
+This section describes how the files of `*.csolution.yml` should be organized to allow the scenarios described above:
+
+Source Directory                  | Content
+:-----------------------------------|:---------------
+`.`                                 | Contains one or more `*.csolution.yml` files that describes an overall application.
+`./<project>`                       | Each project has its own directory
+`./<project>/RTE+<target>`          | Configurable files that are specific to a target have a specific directory.
+`./<project>/RTE`                   | Configurable files that are common to all targets may have a common directory.
```

Comment:
Component may be configurable and applicable to all targets, but user may want to have target specific configurations ? For instance MQTT keep alive interval may differ from target-1 which supports WIFI from target-2 which supports cellular.  

Wouldn't that be simpler to have configuration files always in ./<project>/RTE+<target>
Or do we need a way to specify for each of them, if the configuration is target specific ?

Reply:
Agreed that RTE directory needs some work.